procrastinate/blueprints.py

```diff
@@ -0,0 +1,54 @@
+import functools
+from typing import TYPE_CHECKING, Any, Callable, List, Optional
+
+from procrastinate import jobs, retry
+
+if TYPE_CHECKING:
+    from procrastinate import app, tasks
+
+
+class BluePrint:
```

Comment:
Hm... or could `App` inherit `Blueprint` ? That would solve the problem.

Reply:
A nice advantage of that if it's possible is that `register_blueprint` could be called on an app but also on a blueprint, allowing recursive usage. Or allowing registering tasks from another app.

Reply:
I'm not sure I see the problem.

The process involved in `App.register_blueprint` and `Blueprint.register_blueprint` would be different owing to the fact that the former actually binds tasks to the App, but the latter is creating structures to be bound at a later time.

Allowing a blueprint to be registered in a nested fashion would probably be as simple as:

```python
class Blueprint:
    def __init__(self):
        self._blueprints = []
        
    def register_blueprint(self, blueprint: "Blueprint") -> None:
        self._blueprints.append(blueprint)
```

The `App.register_blueprint` would then be responsible for traversing the tree of nested blueprints.

Is this something for this PR, or a separate one?

Reply:
> I'm not sure I see the problem.

The problem is ensuring that Blueprint.task() and App.task() signatures stay synchronized, but we've gone another route in the other thread.

Yep, if we don't do the recursive this now, I believe another PR will be fine. Also, we'll probably want to include a namespace here as well :D